### DISCUSSION TITLE
Type mismatch when setting up connection

### DISCUSSION THREAD

**Discussion by fernandocanizo** [porsager/postgres#569] [Q&A] [answered]
Hello everybody, I'm trying to centralize my application configuration, so I've made a `config.js` file which exports a `porsager/postgres` configuration object like this:

```javascript
export default {
  postgres: {
    host: process.env.PGHOST ?? 'localhost',
    port: process.env.PGPORT ?? 5432,
    database: process.env.PGDATABASE ?? 'neps',
    username: process.env.PGUSER ?? 'neps',
    password: 'neps',
 }
};
```

The idea is to be able to use the standard Postgresql environment configuration variables. But even when I'm using the nullish coalescing operator, Typescript complains with the following error:

```
src/sql.ts:4:22 - error TS2769: No overload matches this call.
  Overload 1 of 2, '(options?: Options<{}> | undefined): Sql<{}>', gave the following error.
    Argument of type '{ host: string; port: string | number; database: string; username: string; password: string; }' is not assignable to parameter of type 'Options<{}>'.
      Types of property 'port' are incompatible.
        Type 'string | number' is not assignable to type 'number | undefined'.
          Type 'string' is not assignable to type 'number'.
  Overload 2 of 2, '(url: string, options?: Options<{}> | undefined): Sql<{}>', gave the following error.
    Argument of type '{ host: string; port: string | number; database: string; username: string; password: string; }' is not assignable to parameter of type 'string'.
```

The problem goes away if I remove the environment variables, so I understand Typescript is complaining about the possibility of them being `undefined`, however I thought the `??` would solve that.

I understand that this may be a "general Typescript" question, but I preferred to ask here first since it involves specific `porsager/postgres` types too. Also, since I'm building an scaffolder that may be used by others in the future, maybe the community could also jump in and discuss my approach (trying to use `PG*` environment variables).

**Comment by boromisp** [comment, marked as answer]
Actually, typescript complains that:
```
      Types of property 'port' are incompatible.
        Type 'string | number' is not assignable to type 'number | undefined'.
          Type 'string' is not assignable to type 'number'.
```

The last line is the most specific.

In order to satisfy the `Options`  type, you would have to do something like this:
```js
{
  port: parseInt(process.env.PGPORT, 10) || 5432
}
```

**Reply by fernandocanizo** [reply]
Thank you @boromisp ,  I missread the error message.
